code/datums/dna.dm

```diff
@@ -456,6 +458,22 @@ GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_uf_len_by_block())
 	return
 
 /////////////////////////// DNA MOB-PROCS //////////////////////
+/datum/dna/proc/update_body_size()
+	if(!holder)
+		return
+	var/desired_size = GLOB.body_sizes[features["body_size"]]
+
+	if(desired_size == current_body_size)
+		return
+
+	if(!features["body_size"])
+		return
+
+	var/change_multiplier = desired_size / current_body_size
+	var/translate = ((change_multiplier-1) * 32) * 0.5
+	holder.transform = holder.transform.Scale(change_multiplier)
+	holder.transform = holder.transform.Translate(0, translate)
+	current_body_size = desired_size
```

Comment:
The `if(!features["body_size"])` check should be moved up a bit since `features["body_size"]` is being accessed on line 464.
```suggestion
/datum/dna/proc/update_body_size()
	if(!holder || !features["body_size"])
		return
	var/desired_size = GLOB.body_sizes[features["body_size"]]

	if(desired_size == current_body_size)
		return

	var/change_multiplier = desired_size / current_body_size
	var/translate = ((change_multiplier-1) * 32) * 0.5
	holder.transform = holder.transform.Scale(change_multiplier)
	holder.transform = holder.transform.Translate(0, translate)
	current_body_size = desired_size
```